sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/OpenAIClientBuilderTest.java

```diff
@@ -0,0 +1,175 @@
+
+        assertEquals(httpLogOptions, actualHttpLogOptions);
+        assertNotNull(actualHttpLogOptions);
+    }
+
+    @Test
+    void testPipelineSetter() throws NoSuchFieldException, IllegalAccessException {
+        HttpPipeline pipeline = new HttpPipelineBuilder().build();
+        OpenAIClientBuilder builder = new OpenAIClientBuilder().pipeline(pipeline);
+
+        Field pipelineField = OpenAIClientBuilder.class.getDeclaredField("pipeline");
+        pipelineField.setAccessible(true);
+        HttpPipeline actualPipeline = (HttpPipeline) pipelineField.get(builder);
+
+        assertEquals(pipeline, actualPipeline);
+        assertNotNull(actualPipeline);
+    }
+
+    @Test
+    void testBuildClientWithHttpPipeline() throws NoSuchFieldException, IllegalAccessException {
+        HttpPipeline pipeline = new HttpPipelineBuilder().build();
+
+        OpenAIClientBuilder builder = new OpenAIClientBuilder().pipeline(pipeline).endpoint("https://api.openai.com/");
+
+        assertNotNull(builder.buildClient());
+
+        Field pipelineField = OpenAIClientBuilder.class.getDeclaredField("pipeline");
+        pipelineField.setAccessible(true);
+        HttpPipeline actualPipeline = (HttpPipeline) pipelineField.get(builder);
+
+        assertEquals(pipeline, actualPipeline);
+    }
+
+    @Test
+    void testBuildClientWithHttpLogOptions() {
+        HttpLogOptions logOptions = new HttpLogOptions();
+
+        OpenAIClientBuilder builder
+            = new OpenAIClientBuilder().httpLogOptions(logOptions).endpoint("https://api.openai.com/");
+
+        assertNotNull(builder.buildClient());
+    }
+
+    @Test
+    void testBuildClientWithRetryPolicy() {
+        RetryPolicy retryPolicy = new RetryPolicy();
+
+        OpenAIClientBuilder builder
+            = new OpenAIClientBuilder().retryPolicy(retryPolicy).endpoint("https://api.openai.com/");
+
+        assertNotNull(builder.buildClient());
+    }
+
+    @Test
+    void testAddPolicy() {
+        HttpPipelinePolicy customPolicy = (context, next) -> {
+            assertNotNull(context, "The context parameter should not be null");
+            return next.process();
+        };
+
+        OpenAIClientBuilder builder
+            = new OpenAIClientBuilder().addPolicy(customPolicy).endpoint("https://api.openai.com/");
+
+        assertNotNull(builder.buildClient());
+
+        try {
+            Field pipelinePoliciesField = OpenAIClientBuilder.class.getDeclaredField("pipelinePolicies");
+            pipelinePoliciesField.setAccessible(true);
+            List<?> rawList = (List<?>) pipelinePoliciesField.get(builder);
+            List<HttpPipelinePolicy> pipelinePolicies = rawList.stream()
+                .filter(HttpPipelinePolicy.class::isInstance)
+                .map(HttpPipelinePolicy.class::cast)
+                .collect(Collectors.toList());
+
+            assertTrue(pipelinePolicies.contains(customPolicy));
+        } catch (NoSuchFieldException | IllegalAccessException e) {
```

Comment:
I think it's best if we move these to be thrown in the method signature, like you've done for the other tests.